internal/overlord/servstate/manager.go

```diff
@@ -481,6 +485,42 @@ func (m *ServiceManager) CheckFailed(name string) {
 	}
 }
 
+// SetServiceArgs sets the service arguments provided by "pebble run --args"
+// to their respective services. It updates the service command in the top most
+// layer which contains the service in the plan.
```

Comment:
This is a lot closer, but still disagrees with what was suggested in the previous review and during our conversationo. The problem here is that it is still corrupting the plan and modifying something completely unrelated in a way that will be misleading to those inspecting the system. It also means that if someone takes off their own layer later, they'll mistakingly remove the extra arguments unexpectedly.

That's the sort of reason why the suggestion was to _append_ an additional layer that contains the arguments changed, in a clear and clean way.

Reply:
Ah okay. Perhaps I misunderstood a bit. I have updated this in https://github.com/canonical/pebble/pull/191/commits/8e6ef62bef2c5afd5f3bce225a5d0995bc0e0d71. Added a new layer as you suggested.